File: 242_ValidAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        smap = {}
        tmap = {}
        for i in range(len(s)):
            if chr(ord(s[i])) not in smap:
                smap[chr(ord(s[i]))] = 1
            else:
                smap[chr(ord(s[i]))] += 1
                
                
        for i in range(len(t)):
            if chr(ord(t[i])) not in tmap:
                tmap[chr(ord(t[i]))] = 1
            else:
                tmap[chr(ord(t[i]))] += 1
        
        #print(smap)
        #print(tmap)
            
        
        return smap == tmap
        

File: 242_ValidAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        map<char, int> string1;
        map<char, int> string2;
        
        for(int i = 0; i < s.size(); i++)
        {
            string1[s.at(i)] += 1;
        }
        
        for(int i = 0; i < t.size(); i++)
        {
            string2[t.at(i)] += 1;
        }
        
        if(string1 == string2)
        {
            return true;
        }
        else
        {
            return false;
        }
        
    }
};